api/db/index.js

```diff
@@ -0,0 +1,27 @@
+const defaultKnex = require('knex');
+const defaultBookshelf = require('bookshelf');
+const defaultConfig = require('../knexfile');
+
+const user = require('./user');
+const authorization = require('./authorization');
+
+const exportedModels = {};
+
+const setup = (
+  knex = defaultKnex,
+  bookshelf = defaultBookshelf,
+  config = defaultConfig,
+  models = { user, authorization }
+) => {
+  const db = knex(config[process.env.NODE_ENV]);
+  const orm = bookshelf(db);
+
+  Object.keys(models).forEach(modelName => {
+    exportedModels[modelName] = models[modelName](orm);
```

Comment:
I'm confused by this bit -- the `exportedModels` object only gets populated when `setup` is called (which prompts the `forEach` loop) but until then (and in the module.exports), it's just an empty `{}`, is that right?

Reply:
Correct.  It has to modify the existing object because the default export is cached by Node.js the first time it's created.  If you reassign `exportedModels` to a new object, nobody will see it because Node.js has cached off the original value.  If you modify the `exportedModels` object, though, everyone gets it.  Yay, by-reference.  😄 

So basically this is the alternate version of cacheing I described in the other comment.  It only just now occurs to me that I'm doing it two different ways...  Should almost certainly land on just one or the other.

Reply:
Gotcha. Ok, I understand now; makes sense. Do you have a cacheing approach preference?

Reply:
Not really, but now I want to think about it for a little bit before we merge it.  The good news is that we can change it later without much problem.

Reply:
Ah, now I remember.  The idea is that to access models, you'd do `require('../db').models.user` (for example) rather than referencing the model directly.  The models would self-cache just in case someone *did* reference them directly, but the intent is that by going through the `db` module, you could get to them as objects and not have to remember to call them as functions (which I know from experience, I would forget *all the time*).

They're still functions under the hood, though, so that their dependencies can be injected for testing.

I think the whole thing can be cleaned up some more, though.  And it's also not entirely consistent right now, to boot - to get the roles model, for example, you'd need to do `require('../db').models.authorization.roles` and that is _*SUPER*_ non-obvious.

Reply:
Ahh, I see - yeah I like this approach to access models. 👍